src/Core/Services/TypeHelper.cs

```diff
@@ -241,22 +241,16 @@ public static Type GetSystemTypeFromSqlDbType(string sqlDbTypeName)
         }
 
         /// <summary>
-        /// Helper method to find the DbType for the given datetime data types in Sql Server.
+        /// Helper method to get the DbType corresponding to the given SqlDbType.
         /// </summary>
-        /// <param name="sqlDbTypeName">Data type of the datetime column in Sql Server.</param>
-        /// <returns>Corresponding DbType for the column.</returns>
-        /// <exception cref="DataApiBuilderException"></exception>
-        public static DbType GetDbTypeFromSqlDbDateTimeType(string sqlDbTypeName)
+        public static DbType? GetDbTypeFromSqlDbType(SqlDbType sqlDbType)
         {
-            if (_timeSqlDbTypeToDbType.TryGetValue(sqlDbTypeName, out DbType value))
+            if (_sqlDbTypeToDbType.TryGetValue(sqlDbType, out DbType dbType))
             {
-                return value;
+                return dbType;
             }
 
-            throw new DataApiBuilderException(
-                message: $"Unexpected datetime data type: {sqlDbTypeName} encountered.",
-                statusCode: HttpStatusCode.ServiceUnavailable,
-                subStatusCode: DataApiBuilderException.SubStatusCodes.UnexpectedError);
+            return null;
```

Comment:
This null should not be hit ideally now. But if it does, it means there are more mssql datatypes  which map to DateTime/DateTimeOffset system types other than date,datetime,datetime2,smalldatetime,datetimeoffset. I have covered all the datatypes as of now, but if this code block is hit, probably MsSql started supported a new datatype which we don't currently have a mapping for. Returning null instead of exception prevents DAB from handling that unexpected case.

Reply:
By returning null now, aren't we silently hiding the case? Would it be better to throw an exception either here or in the caller to identify whenever in future a new date time type is added?

Reply:
Yeah, I think that makes sense.

Reply:
We will log a warning to just know that an additional datatype was added to MsSql which maps to .NET type of datetime.